Teamleder page og endringer relatert til dette

tbhansen opened the pull request:
Inneholder også rettelse av formateringsfeil.

tbhansen committed:
"Problemer med DeleteBrukere controller"

tbhansen committed:
DeleteBrukere API fungerer.

tbhansen committed:
Linjeoppsett

tbhansen committed:
Merge branch 'IS-200-G7:master' into TeamlederPage

tbhansen committed:
Merge pull request #1 from tbhansen/TeamlederPage

Teamleder page

tbhansen committed:
Lagt til funksjon for å vise teamet til den som er innlogget. Brukt sql joins for å vise info fra annet table

tbhansen committed:
Merge pull request #2 from tbhansen/TeamlederAndControllers

SessionId i LocalStorage. Sql joins for fremvisning av data.

tbhansen committed:
Endringer i addBrukerToTeam API. Løste constraint feil

tbhansen committed:
Rettet opp i div formatering

tbhansen committed:
Delete global.json

sdushantha commented on PDSA_System/Client/Pages/Login.razor:
```diff
@@ -51,6 +51,9 @@
                 var token = task.Result.Content.ReadAsStringAsync().Result;
 
                 LocalStorage.SetItemAsStringAsync("token", token);
+                //Linje nedenfor brukes av TeamlederPage. Må endres manuelt foreløpig. Teams må inn i local storage.
+                //Endre string data for å endre team.
+                LocalStorage.SetItemAsStringAsync("teamId", "4");
```
Skal dette være her?

tbhansen replied:
Ja. Dette setter TeamId i session. Kan forklare nærmere senere.

sdushantha replied:
Men hva skjer hvis min TeamID er 2 da? Tallet 4 er jo hardkodet

tbhansen replied:
For å implementere dette må det teamid inn i tokenet som opprettes ved pålogging. Endringer må gjøres på authcontroller og Login razorpage.

Er på litt dypt vann her, så mulig terminologien min er noe feil her.

menymo replied:
> For å implementere dette må det teamid inn i tokenet som opprettes ved pålogging. Endringer må gjøres på authcontroller og Login razorpage.

Jeg hadde ikke satt det inn i token, da en endring av dette vil invalidere det og logge ut brukeren. Gjør det heller mulig for bruker å se en enkel liste over team, og legg til en knapp for å endre team, så vil den redigere på den nye verdien som du har opprettet

tbhansen committed:
Update TeamlederPage.razor

Lagt til AuthorizeView igjen. Fjernet en kommentar som ikke var gyldig lenger, og unødvendig innhold i modal content.

menymo reviewed (approved):
Funker bra, bare å merge

Vrolf commented on PDSA_System/Client/Pages/TeamlederPage.razor:
```diff
@@ -0,0 +1,179 @@
+@inject HttpClient Http
+@inject NavigationManager NavigationManager
+@inject ILocalStorageService LocalStorage
+<AuthorizeView>
```
legg til `Roles ="teamleder"` i Authenticate View

tbhansen replied:
Takk igjen:)

tbhansen committed:
Lagt inn <AuthorizeView Roles="teamleder">

tbhansen committed:
Update TeamlederPage.razor

Endret fra to << til en << i Authorize view.

menymo committed:
Merge branch 'master' into TeamlederAddBrukerIssue
